### ISSUE TITLE
[OpenGL 4.4] Timing issue (Multi Draw Arrays)

### ISSUE THREAD

**Issue by Nielsbishere** [baldurk/renderdoc#1022]
## Description

Everytime I have a capture that contains a multi draw arrays command, RenderDoc fails to correctly estimate the time every frame takes. This means that every capture I profile is inaccurate for the engine I'm working on. The weird thing is that every time I click the 'time' button, the time changes and eventually reaches 13ms instead of the original 115ms (on a 960M). A team member also tried it and for him it changed from 400ms to 7ms (on a 970M), this also happens on my desktop (1060) (but I haven't timed it).

I don't know exactly how RenderDoc works, but from viewing this; I am pretty sure that you execute one (or several) command list(s) per frame. This means that either the indirect calls have a performance hit in RenderDoc, they require setup time or that they are indeed slower for the first frames. If it is slower the first frames; you could perform multiple timings (automatically) to ensure that the results you show are accurate (or add an extra button that does that).

## Repro steps

Capture: https://ufile.io/07cqz
Application: (Not available, due to NDA)

## Environment

RenderDoc build: https://github.com/baldurk/renderdoc/commit/569f11b7577ba77458f41d81116b545cc82971c6
Operating System: Windows 10 x64
API: OpenGL 4.4
GPU: GTX 960M and one with 970M

**Comment by baldurk** [comment]
I don't think this is specific to MDI, the timings in general are known to be unreliable - see e.g. #65 although this has come up often in various ways. I've seen the reverse situation - where the first timing is mostly accurate but repeating after that drifts away (in either direction - up or down). The situation is worse on older APIs like OpenGL where it's difficult to control exactly what's submitted, Vulkan is a bit better but it's still not great. You mention submission in one or several command lists but OpenGL has no way of controlling that.

You might have better luck if you have access to an AMD GPU and use their GPU counters, but I don't know for sure. If all you're interested in is whole-frame timings then I'd recommend you time yourself in your application as you should be able to get more reliable data.

The bottom line is that RenderDoc is *not* a profiler and isn't intended to support profiling. It's a large amount of work to do properly and it's not work that's currently planned.
